server/issue.go

```diff
@@ -256,7 +256,30 @@ func (s *Server) registerIssueRoutes(g *echo.Group) {
 }
 
 func (s *Server) createIssue(ctx context.Context, issueCreate *api.IssueCreate, creatorID int) (*api.Issue, error) {
-	pipeline, err := s.createPipelineFromIssue(ctx, issueCreate, creatorID)
+	// Run pre-condition check first to make sure all tasks are valid, otherwise we will create partial pipelines
+	// since we are not creating pipeline/stage list/task list in a single transaction.
+	// We may still run into this issue when we actually create those pipeline/stage list/task list, however, that's
+	// quite unlikely so we will live with it for now.
+	pipelineCreate, err := s.getPipelineCreate(ctx, issueCreate)
+	if err != nil {
+		return nil, err
+	}
+
+	if !issueCreate.ValidateOnly {
```

Comment:
let's add a comment explaining why not check assignee when ValidateOnly is true

Reply:
Done. PTAL